src/javaharness/java/arcs/api/ArcsEnvironment.java

```diff
@@ -13,11 +11,6 @@
     void onData(String arcId, String data);
   }
 
-  /** Called by ArcsEnvironment when the Arcs runtime is ready. */
-  interface ReadyListener {
```

Comment:
I'd rather we don't delete anything (at least without confirming with Ray and Scott first) - there might be some older demos that depend on this and might still be useful. Maybe add deprecated warning logs instead.

Reply:
On this particular thing, if nothing is calling it now, I do think we likely need something. This was, at least a while back, how Android knew when the shell was booted up and available to respond to requests. @cromwellian

Reply:
Ah, interesting, actually I could use one of these!
I temporarily just added the functionality I needed to the DeviceClient `ready` message case, because these ReadyListeners can be added, but are never called, probably just got lost in between refactorings...

Reply:
yes, the shell is sending the `ready` message,  but the `readyListeners` are not called, where they should be called, i think: https://github.com/PolymerLabs/arcs/blob/master/src/javaharness/java/arcs/api/DeviceClientImpl.java#L48

Reply:
Reverted